### PR TITLE
port `bcftools-stats` to nf-test

### LINKED ISSUE #5255: migrate `bcftools/stats` to nf-test

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #5255 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/bcftools/stats/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.version).match("targets_version") },
+                { assert snapshot(file(process.out.stats.get(0).get(1)).readLines()[0..5]).match() },
+            )
+        }
+
+    }
```
Comment (itrujnara) [inline modules/nf-core/bcftools/stats/tests/main.nf.test:93]: I don't see a test for samples, is this intentional?

**Review by itrujnara — COMMENTED** [review]
Looks legit, but it seems one test might be missing. Let me know if it's intentional.

File: modules/nf-core/bcftools/stats/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    tag "bcftools"
+    tag "bcftools/stats"
+
+    test("sarscov2 - vcf_gz") {
+
+        when {
+            process {
+                """
+                input[0] = [ [ id:'test' ], // meta map
+                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
```
Comment (jonasscheid) [inline modules/nf-core/bcftools/stats/tests/main.nf.test:18]: Can you update these paths according to e.g. this [PR](https://github.com/nf-core/modules/pull/5281/files)?
That way is preferred to run the tests properly on the cloud mirror (afaik)

File: modules/nf-core/bcftools/stats/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    test("sarscov2 - vcf_gz - regions") {
+
+        when {
+            process {
+                """
+                input[0] = [ [ id:'regions_test' ], // meta map
+                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
+                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)]
+                input[1] = [ [id:'regions_test'],
+                            file(params.test_data['sarscov2']['illumina']['test3_vcf_gz'], checkIfExists: true) ]
```
Comment (jonasscheid) [inline modules/nf-core/bcftools/stats/tests/main.nf.test:48]: Same here

File: modules/nf-core/bcftools/stats/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+        when {
+            process {
+                """
+                input[0] = [ [ id:'targets_test' ], // meta map
+                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
+                            [] ]
+                input[1] = [ [], [] ]
+                input[2] = [ [id:'targets_test'],
+                            file(params.test_data['sarscov2']['illumina']['test2_vcf_targets_tsv_gz'], checkIfExists: true) ]
```
Comment (jonasscheid) [inline modules/nf-core/bcftools/stats/tests/main.nf.test:77]: and here

File: modules/nf-core/bcftools/stats/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            process {
+                """
+                input[0] = [ [ id:'exon_test' ], // meta map
+                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
+                            [] ]
+                input[1] = [ [], [] ]
+                input[2] = [ [], [] ]
+                input[3] = [ [], [] ]
+                input[4] = [ [id: "exon_test"],
+                            file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/bcftools/stats/exons.tsv.gz") ]
```
Comment (jonasscheid) [inline modules/nf-core/bcftools/stats/tests/main.nf.test:107]: This path seems to get depricated soon if it is in `delete_me` right? You might consider to move it again to the proper location if you want to use it..

File: modules/nf-core/bcftools/stats/tests/nextflow.config [hunk 6]
```diff
@@ -0,0 +1,2 @@
+process {
+}
```
Comment (jonasscheid) [inline modules/nf-core/bcftools/stats/tests/nextflow.config:2]: Is this file needed?

**Review by jonasscheid — CHANGES_REQUESTED** [review]
Nice one 🚀 A couple of comments before merging

File: modules/nf-core/bcftools/stats/tests/main.nf.test (hunk 4 above)
Comment (jonasscheid) [reply modules/nf-core/bcftools/stats/tests/main.nf.test:77]: and basically in all following paths 😁

File: modules/nf-core/bcftools/stats/tests/main.nf.test (hunk 1 above)
Comment (famosab) [reply modules/nf-core/bcftools/stats/tests/main.nf.test:93]: I just ported what was there in the pytest and that one was missing there.

File: modules/nf-core/bcftools/stats/tests/nextflow.config (hunk 6 above)
Comment (famosab) [reply modules/nf-core/bcftools/stats/tests/nextflow.config:2]: Probably not? I just ported from pytest so that might be why its there.

File: modules/nf-core/bcftools/stats/tests/main.nf.test (hunk 2 above)
Comment (famosab) [reply modules/nf-core/bcftools/stats/tests/main.nf.test:18]: So we stop using this dictionary? https://github.com/nf-core/modules/blob/master/tests/config/test_data.config

File: modules/nf-core/bcftools/stats/tests/main.nf.test (hunk 5 above)
Comment (famosab) [reply modules/nf-core/bcftools/stats/tests/main.nf.test:107]: Should I open a PR to test-datasets then? Its a bit weird that it was moved there and still used in the pytest.

Comment (maxulysse) [reply modules/nf-core/bcftools/stats/tests/main.nf.test:108]:
```suggestion
                            file(params.modules_testdata_base_path + 'delete_me/bcftools/stats/exons.tsv.gz', checkIfExists: true) ]
```

**Review by maxulysse — APPROVED** [review]

**Review by jonasscheid — APPROVED** [review]
LGTM!
